### PR TITLE
boards: shields: Adding support for the waveshare pico ups-b shield

### PR DESCRIPTION
**Pull request by joeker64**
Adding documentation and support for the Waveshare pico ups-b shield. I have also added a alternate nodelabel for the Raspberry Pi Pico i2c 'pico_i2c'

### CODE REVIEW

File: boards/shields/waveshare_ups/Kconfig.shield [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Copyright (c) 2023 Joseph Yates
```
Comment (soburi) [inline boards/shields/waveshare_ups/Kconfig.shield:1]: Please add space after `#`.

File: boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023 Joseph Yates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&pinctrl{
+	i2c1_default: i2c1_default{
```
Comment (soburi) [inline boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay:8]: Please add space before `{`.

File: boards/arm/rpi_pico/rpi_pico-common.dtsi [hunk 3]
```diff
@@ -145,3 +145,4 @@ zephyr_udc0: &usbd {
 };
 
 pico_spi: &spi0 {};
+pico_i2c: &i2c1 {};
```
Comment (soburi) [inline boards/arm/rpi_pico/rpi_pico-common.dtsi:148]: The i2c1 is not defaultly available.
It should move to .overlay file.

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/arm/rpi_pico/rpi_pico-common.dtsi (hunk 3 above)
Comment (joeker64) [reply boards/arm/rpi_pico/rpi_pico-common.dtsi:148]: Thanks, will make these changes this evening :)

**Review by soburi — CHANGES_REQUESTED** [review]

File: boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay (hunk 2 above)
Comment (joeker64) [reply boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay:8]: Okay :))

**Review by soburi — DISMISSED** [review]

File: boards/shields/waveshare_ups/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
+.. _waveshare_pico_ups_b_shield:
+
+Waveshare Pico UPS-B shield
+#################################
```
Comment (erwango) [inline boards/shields/waveshare_ups/doc/index.rst:4]: Please fix under line lenght

File: boards/shields/waveshare_ups/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+Overview
```
Comment (erwango) [inline boards/shields/waveshare_ups/doc/index.rst:6]: Ditto

File: boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+		group1 {
+			pinmux = <I2C1_SDA_P6>, <I2C1_SCL_P7>;
+			input-enable;
+                        input-schmitt-enable;
+			bias-pull-up;
+		};
+	};
+};
+
+pico_i2c1: &i2c1 {};
```
Comment (erwango) [inline boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay:18]: This is a board setting. This definition may likely not fit on all boards.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution, but this doesn't fit to a shield description.
A correct shield description should be compatible with several boards through the use of a common connector.
I don't see a connector in use here, so this can only be compatible with a unique board and hence doesn't match shield definiton.

See https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-compatibility

**Comment by joeker64** [comment]
> Thanks for this contribution, but this doesn't fit to a shield description. A correct shield description should be compatible with several boards through the use of a common connector. I don't see a connector in use here, so this can only be compatible with a unique board and hence doesn't match shield definiton.
> 
> See https://docs.zephyrproject.org/latest/hardware/porting/shields.html#board-compatibility

The common connector would be I2C? The pico can support having multiple I2C devices with I2C0 & I2C1. The shield is setup to use the I2C1 pins. I was going to add support for I2C1 in the default pico_dts but found this commit so I included it in the shield overlay:

https://github.com/zephyrproject-rtos/zephyr/commit/9eb7ee6ed8308bcee88bc87b13a7c5991166e2bf

File: boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay (hunk 6 above)
Comment (soburi) [inline boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay:18]: Could you consider to create `boards/shields/waveshare_ups/boards/rpi_pico.overlay`.

Comment (joeker64) [reply boards/shields/waveshare_ups/waveshare_pico_ups_b.overlay:18]: Yes, will get on that now :)

File: boards/arm/rpi_pico/rpi_pico-common.dtsi [hunk 7]
```diff
@@ -145,3 +145,5 @@ zephyr_udc0: &usbd {
 };
 
 pico_spi: &spi0 {};
+pico_i2c0: &i2c0 {};
+pico_i2c1: &i2c1 {};
```
Comment (soburi) [inline boards/arm/rpi_pico/rpi_pico-common.dtsi:149]: `i2c1` not always available.

**Review by soburi — CHANGES_REQUESTED** [review]

**Review by soburi — DISMISSED** [review]
Please fix pointing out by @erwango.
No additional comments from me.

**Comment by joeker64** [comment]
I added in back support for I2C1 but set it too disabled as default. This was so I didn't have to write i2c1 device logic in the board overlay in the shield directory.

**Comment by joeker64** [comment]
Is there anything I need to do for this to be merged?

**Comment by soburi** [comment]
> Is there anything I need to do for this to be merged?

Please fix pointing out by @erwango.

https://github.com/zephyrproject-rtos/zephyr/pull/60384#discussion_r1277219803
https://github.com/zephyrproject-rtos/zephyr/pull/60384#discussion_r1277219896

File: boards/shields/waveshare_ups/doc/index.rst (hunk 4 above)
Comment (soburi) [inline boards/shields/waveshare_ups/doc/index.rst:4]:
```suggestion
######################
```

**Review by soburi — CHANGES_REQUESTED** [review]
Section header underline must be same as title length.

**Comment by soburi** [comment]
@joeker64 

`Commit suggestion` is not allowed by CI check.
Please rebase and squash these.
And one point, my first correction is wrong.
Please correct it to the correct number of "#".

File: boards/shields/waveshare_ups/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+***************************
```
[the remark this answers is not in the document]
Comment (soburi) [reply boards/shields/waveshare_ups/doc/index.rst:7]: Here is `********`.
As same length of `Overview`.

**Review by soburi — APPROVED** [review]

**Comment by soburi** [comment]
@erwango 

Could you take a look, again?

**Review by erwango — APPROVED** [review]

**Pull request merged**